metals/src/main/scala/scala/meta/internal/metals/codeactions/AddingBracesCodeAction.scala

```diff
@@ -0,0 +1,478 @@
+      codeActionSubjectTitle = codeActionSubjectTitle,
+      maybeEndMarkerPos = maybeGetEndMarkerPos(blockHolder)
+    )
+  }
+
+  /**
+   * @param blockHolder      the trees which can only be braced if the type of a branch of
+   *                         them is `Term.Block` and that block is braced.
+   * @param path             the path to the file containing the tree. It is used in
+   *                         contstructing the TextEdit of the code action
+   * @param document
+   * @param blockEmbraceable the branch of the main tree which can potentially have
+   *                         the type `Term.Block`; so can be braced.
+   * @param codeActionSubjectTitle
+   * @return
+   */
+  private def createCodeActionForPotentialBlockHolder(
+      blockHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      blockEmbraceable: Term,
+      codeActionSubjectTitle: String,
+      bracelessStart: String = ""
+  ): Option[l.CodeAction] = {
+    val indentation =
+      getIndentationForPositionInDocument(blockHolder.pos, document)
+
+    if (!isBlockEmbraceableBraced(blockEmbraceable)) {
+      createCodeActionToTakePotentialBlockHolderBraceful(
+        blockHolder,
+        path,
+        document,
+        blockEmbraceable,
+        codeActionSubjectTitle,
+        indentation,
+        bracelessStart
+      )
+    } else None
+  }
+
+  private def maybeGetEndMarkerPos(tree: Tree): Option[Position] =
+    tree.parent
+      .flatMap(parent =>
+        parent.children.dropWhile(_ != tree).tail.headOption.collectFirst {
+          case endMarker: Term.EndMarker => endMarker
+        }
+      )
+      .map(_.pos)
+
+  private def createCodeActionForTemplateHolder(
+      templateHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      templ: Template,
+      codeActionSubjectTitle: String
+  ): Option[l.CodeAction] = {
+    val indentation =
+      getIndentationForPositionInDocument(templateHolder.pos, document)
+    val expectedBraceStartPos = util
+      .Try {
+        val lastInit = templ.inits.maxBy(init => init.pos.end)
+        lastInit.pos.end
+      }
+      .getOrElse(templ.pos.start)
+    if (!hasBraces(templ)) {
+      createCodeActionToTakeTemplateHolderBraceful(
+        templateHolder,
+        path,
+        document,
+        templ,
+        codeActionSubjectTitle,
+        expectedBraceStartPos,
+        indentation
+      )
+    } else None
+  }
+
+  private def createCodeActionToTakeTemplateHolderBraceful(
+      templateHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      templ: Template,
+      codeActionSubjectTitle: String,
+      expectedBraceStartPos: Int,
+      indentation: String
+  ): Option[l.CodeAction] = {
+
+    for {
+      colonPose <- templ.tokens
+        .find(token =>
+          token.text == ":" && token.pos.start >= expectedBraceStartPos
+        )
+        .map(_.pos.toLSP.getStart)
+    } yield createCodeActionForGoingBraceful(
```

Comment:
and also the `createCodeActionForGoingBraceful` is called here. If it is removed, the code of it should be duplicated in these two places.